qt-models/messagehandlermodel.cpp

```diff
@@ -51,6 +51,15 @@ void MessageHandlerModel::addLog(QtMsgType type, const QString& message)
 #endif
 }
 
+void MessageHandlerModel::copyLogToString(QString &copyString)
+{
+	QVector<MessageData>::iterator i = MessageHandlerModel::self()->m_data.begin();
```

Comment:
Seems overly complicated:
```
for(const MessageData &data: m_data)
    copyString += data.message;
```

Reply:
For silly reasons (which I am sure a better developer with more Qt/QML understanding than I have) the log is currently in a model... so I think this IS the easiest way to do it.
But maybe someone can simply turn this into a string that is displayed?
IIRC the problem was that I couldn't get this to "line break" and scroll the way I wanted and so I turned it into a view of a Model instead...

Reply:
I wasn't commenting on the model thing. My point was that the loop can be written in a much simpler way.

Reply:
I happend to agree with @bstoeger and have changed it. I am just a slave of my past, where this kind of iterator hiding was forbidden.

Reply:
> I am just a slave of my past, where this kind of iterator hiding was forbidden.

That's an exceptionally peculiar policy.